README.md

```diff
@@ -179,3 +179,68 @@ spec:
  nodeSelector:
    master: eno3
 ```
+
+### Customized Injections
+
+Customized injections allows user to define additinal injections (besides what's supported in NRI, such as ResourceName, Downward API volumes etc) in Kubernetes ConfigMap and request additional injection for individual pod based on pod label. Currently customized injection only support injecting pod annotations.
+
+In order to use this feature, user needs to create the customized injection ConfigMap with name `nri-customized-injections` in namespace `kube-system`. The data entry in ConfigMap is in the format of key:value pair. Key is a user defined label that will be used to match with pod labels, Value is the actual injection that will be applied to pod manifest. NRI would listen to the creation/update/deletion of this ConfigMap and update its internal data structure every 30 seconds so that subsquential creation of pods will be evaluated against the latest customized injections.
```

Comment:
missing info here on the format of the injection, there is just an example below but i think a more formal definition is needed.

is it in the format as defined by https://tools.ietf.org/html/rfc6902 ?

Reply:
In addition you should also state that these user defined patches take precedence over built-in patch operations performed by NRI

Reply:
The order of injection may depend on what customized patch are configured by user. For current annotation, it takes precedence then built-in patch, because there is a chance that annotation may contain net-attach-def definition, that would impact the injeciton of resourceName. I'll add the precedence info.

The format is json string, within which, the value structure is an internal data structure defined at: https://github.com/k8snetworkplumbingwg/network-resources-injector/blob/master/pkg/webhook/webhook.go#L43

Reply:
> The order of injection may depend on what customized patch are configured by user. For current annotation, it takes precedence then built-in patch, because there is a chance that annotation may contain net-attach-def definition, that would impact the injeciton of resourceName. I'll add the precedence info.

Great, Thanks !


> The format is json string, within which, the value structure is an internal data structure defined at:

So The README section should contain this information IMO (the format) as when a user goes through this section he should understand how to write these custom patches which does match RFC6902

Reply:
Added RFC6902 reference link.